### PR TITLE
DOCS-178 Architecture Overview

### PR DESCRIPTION
**Pull request by angelasimms**
Added/updated descriptions of Dedicated and Serverless with some design details.

### CODE REVIEW

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 1]
```diff
@@ -1,9 +1,24 @@
 = Dedicated Clusters
-:description: {hazelcast-cloud} Dedicated gives you fine-grained control over the hardware and resources available to your cluster. It's your responsibility to scale your clusters when you need more resources. Dedicated clusters also offer strict container isolation for projects that require enhanced data security.
+:description: {hazelcast-cloud} Dedicated is a fully-managed cloud service that allows you to maintain control over the hardware, resources, and deployment options that are available to your cluster. Dedicated means you get specific servers (cloud instances) all to yourself.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:2]:
```suggestion
:description: {hazelcast-cloud} Dedicated is a managed cloud service that allows you to maintain control over the hardware, resources, and deployment options that are available to your cluster. Dedicated means you get specific servers (cloud instances) all to yourself. 
```

File: docs/modules/ROOT/pages/serverless-cluster.adoc [hunk 2]
```diff
@@ -1,10 +1,12 @@
 = Serverless Clusters
-:description: {hazelcast-cloud} Serverless clusters dynamically scale, depending on the amount of resources you use. Serverless offers a simple pricing model based on storage consumption with the option of creating a cluster for free, without a payment card.
+:description: {hazelcast-cloud} Serverless is a fully-managed cloud service that offers a simple pay-as-you-go pricing model for clusters. Serverless clusters scale dynamically to provide the resources that your application needs, and you pay for the resources that your application consumes. You can start by creating a cluster for free, without a payment card.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/serverless-cluster.adoc:2]:
```suggestion
:description: {hazelcast-cloud} Serverless is a managed cloud service that offers a pay-as-you-go pricing model. Serverless clusters scale to provide the resources that your application needs. You pay only for the resources that your application consumes.
```

File: docs/modules/ROOT/pages/serverless-cluster.adoc [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
 :toc-levels: 3
 :page-serverless: true
 
 {description}
 
+Serverless means that Hazelcast manages the cloud infrastructure for you. Each Serverless cluster is an independent deployment of Hazelcast running in isolation on a Kubernetes container. This design guarantees resource isolation (memory and CPU), and prevents resource stealing, as well as providing isolated network access.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/serverless-cluster.adoc:8]:
```suggestion
Serverless means that Hazelcast manages the cloud infrastructure for you. Each Serverless cluster is an independent deployment of Hazelcast in a Kubernetes container. This design guarantees resource isolation, prevents resource stealing, and provides isolated network access.
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 4]
```diff
@@ -1,9 +1,24 @@
 = Dedicated Clusters
-:description: {hazelcast-cloud} Dedicated gives you fine-grained control over the hardware and resources available to your cluster. It's your responsibility to scale your clusters when you need more resources. Dedicated clusters also offer strict container isolation for projects that require enhanced data security.
+:description: {hazelcast-cloud} Dedicated is a fully-managed cloud service that allows you to maintain control over the hardware, resources, and deployment options that are available to your cluster. Dedicated means you get specific servers (cloud instances) all to yourself. 
 :page-dedicated: true
 
 {description}
 
+You can design a Hazelcast topology that meets your specific compliance and service design needs.
+
+[discrete]
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:9]:
```suggestion
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 5]
```diff
@@ -1,9 +1,24 @@
...
-:description: {hazelcast-cloud} Dedicated gives you fine-grained control over the hardware and resources available to your cluster. It's your responsibility to scale your clusters when you need more resources. Dedicated clusters also offer strict container isolation for projects that require enhanced data security.
+:description: {hazelcast-cloud} Dedicated is a fully-managed cloud service that allows you to maintain control over the hardware, resources, and deployment options that are available to your cluster. Dedicated means you get specific servers (cloud instances) all to yourself. 
 :page-dedicated: true
 
 {description}
 
+You can design a Hazelcast topology that meets your specific compliance and service design needs.
+
+[discrete] 
+==== Data and Security Regulations
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:10]:
```suggestion
- Data and security regulations
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 6]
```diff
@@ -1,9 +1,24 @@
...
+- Deploy your clusters to specific regions to comply with data residency laws.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:11]:
```suggestion
** Deploy your clusters to specific regions to comply with data residency laws.
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 7]
```diff
@@ -1,9 +1,24 @@
...
+- Benefit from strict container isolation for projects that require enhanced security. {hazelcast-cloud} Dedicated is SOC 2 and ISO 27001 compliant.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:12]:
```suggestion
** Benefit from strict container isolation for projects that require enhanced security. {hazelcast-cloud} Dedicated is SOC 2 and ISO 27001 compliant.
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
+
+[discrete]
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:14]:
```suggestion
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 9]
```diff
@@ -1,9 +1,24 @@
...
 
+You can design a Hazelcast topology that meets your specific compliance and service design needs.
+
+[discrete] 
+==== Data and Security Regulations
+- Deploy your clusters to specific regions to comply with data residency laws.
+- Benefit from strict container isolation for projects that require enhanced security. {hazelcast-cloud} Dedicated is SOC 2 and ISO 27001 compliant.
+
+[discrete] 
+==== Service Design
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:15]:
```suggestion
- Service design
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 10]
```diff
@@ -1,9 +1,24 @@
...
+- Specify the memory capacity and number of nodes for all provisioned clusters. You decide when to scale your clusters if they need more resources.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:16]:
```suggestion
** Specify the memory capacity and the number of nodes for all provisioned clusters. You decide when to scale your clusters if they need more resources.  
```

Need to check if 'nodes' refers to Kubernetes nodes or Hazelcast members.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 11]
```diff
@@ -1,9 +1,24 @@
...
+
+[discrete] 
+==== Data and Security Regulations
+- Deploy your clusters to specific regions to comply with data residency laws.
+- Benefit from strict container isolation for projects that require enhanced security. {hazelcast-cloud} Dedicated is SOC 2 and ISO 27001 compliant.
+
+[discrete] 
+==== Service Design
+- Specify the memory capacity and number of nodes for all provisioned clusters. You decide when to scale your clusters if they need more resources.  
+- Deploy clusters to specific regions or Cloud providers as part of your organization's Cloud strategy.  To reduce network latency, you can deploy clusters in regions that are closest to the location of your users.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:17]:
```suggestion
** Deploy clusters to specific regions or cloud providers as part of your organization's cloud strategy.  To reduce network latency, you can deploy clusters in regions that are closest to your users' locations.
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 12]
```diff
@@ -1,9 +1,24 @@
...
+- Benefit from Hazelcast support and predictable monthly billing.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:18]:
```suggestion
** Benefit from Hazelcast support and predictable monthly billing.
```

**Review by JakeSCahill — COMMENTED** [review]
Great additions. I have a few suggestions and one point that need clarification regarding nodes vs members.

**Comment by kubamarchwicki** [comment]
I'm walking the overview document and this caught my attention:

> Production Serverless clusters automatically scale up and down

technically we are scaling out and scaling in (we are not making individual members bigger, we are adding new members). That doesn't change the overall picture, but just pointing out it's a different thing.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 1 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:2]: Updated as requested.

File: docs/modules/ROOT/pages/serverless-cluster.adoc (hunk 2 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/serverless-cluster.adoc:2]: Updated as requested.

File: docs/modules/ROOT/pages/serverless-cluster.adoc (hunk 3 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/serverless-cluster.adoc:8]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 4 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:9]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 5 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:10]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 6 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:11]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 7 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:12]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 8 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:14]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 9 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:15]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 10 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:16]: @kubamarchwicki. Could you confirm the details above?

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 11 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:17]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 12 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:18]: Updated as requested.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 10 above)
Comment (kubamarchwicki) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc]: Yes. Hazelcast has members - cluster even brags about it in the logs

```
Members {size:1, ver:1} [
	Member [10.25.131.35]:5701 - bb968318-fec7-4b61-a962-7579a92d9f97 this
]
```

File: docs/modules/ROOT/pages/scale-up-down.adoc [hunk 13]
```diff
@@ -1,20 +1,22 @@
 = Scaling a Dedicated Cluster
-:description: Scaling clusters is useful when you need to scale up for more memory or when you aren't using as much memory as you thought and want to scale down to save on the monthly costs. You can easily scale Dedicated clusters up or down, using a sliding scale that adds or removes cluster members.
+:description: Scaling clusters is useful when you need to increase available memory to handle higher than normal workloads. You can also control monthly costs by reducing memory when you expect lower loads. 
 :page-dedicated: true
 
 {description}
 
+Dedicated clusters scale horizontally, which means that you can adjust the resources available to your application, such as memory, by adding members (scaling out) or removing members (scaling in). Each dedicated cluster is a network of a minimum of three instances or _members_, with a fixed set of resources allocated to each one.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/scale-up-down.adoc:7]:
```suggestion
Dedicated clusters scale horizontally, which means that you can adjust the resources available to your application, such as memory, by adding members (scaling out) or removing members (scaling in). Each Dedicated cluster is a network of at least three instances or _members_, with a fixed set of resources allocated to each one.
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc [hunk 14]
```diff
@@ -1,9 +1,23 @@
...
 
+You can design a Hazelcast topology that meets your specific compliance and service design needs.
+
+- Data and security regulations
+** Deploy your clusters to specific regions to comply with data residency laws.
+** Benefit from strict container isolation for projects that require enhanced security. {hazelcast-cloud} Dedicated is SOC 2 and ISO 27001 compliant.
+
+
+- Service design
+** Specify the memory capacity and the number of members for all provisioned clusters. You decide when to xref:scale-up-down.adoc[scale your clusters] if they need more resources.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/dedicated-cluster.adoc:15]:
```suggestion
** Specify the number of members in your clusters as well as their memory capacity. You decide when to xref:scale-up-down.adoc[scale your clusters].  
```

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 10 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc]: Added explanation to make it clearer what we mean by scaling in and out.

File: docs/modules/ROOT/pages/dedicated-cluster.adoc (hunk 14 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/dedicated-cluster.adoc:15]: Updated as requested.

File: docs/modules/ROOT/pages/scale-up-down.adoc (hunk 13 above)
Comment (angelasimms) [reply docs/modules/ROOT/pages/scale-up-down.adoc:7]: Updated as requested.

**Review by JakeSCahill — APPROVED** [review]

**Review by kubamarchwicki — APPROVED** [review]

**Pull request merged**